examples/advanced/scan.ts

```diff
@@ -0,0 +1,65 @@
+/**
+ * This example demonstrates how to scan the market for providers that meet specific requirements.
+ */
+import { GolemNetwork, ScanOptions } from "@golem-sdk/golem-js";
+import { last, map, scan, takeUntil, tap, timer } from "rxjs";
+
+// What providers are we looking for?
+const scanOptions: ScanOptions = {
+  // fairly powerful machine but not too powerful
+  workload: {
+    engine: "vm",
+    minCpuCores: 4,
+    maxCpuCores: 16,
+    minMemGib: 4,
+    maxMemGib: 8,
+    capabilities: ["vpn"],
+    minStorageGib: 16,
+  },
+  // let's look at mainnet providers only
+  payment: {
+    network: "polygon",
+  },
+};
+
+(async () => {
+  const glm = new GolemNetwork();
+  await glm.connect();
+  const spec = glm.market.buildScanSpecification(scanOptions);
+
+  // For advanced users: you can also add constraints manually:
+  // spec.constraints.push("(golem.node.id.name=my-favorite-provider)");
+
+  const SCAN_DURATION_MS = 10_000;
+
+  console.log(`Scanning for ${SCAN_DURATION_MS / 1000} seconds...`);
+  glm.market
+    .scan(spec)
+    .pipe(
+      tap((scannedOffer) => {
+        console.log("Found offer from", scannedOffer.provider.name);
+      }),
+      // calculate the cost of an hour of work
+      map(
+        (scannedOffer) =>
```

Comment:
The proposal has the OfferProposal.getEstimatedCost method, which actually has a bug in it.

My proposal is to kill two birds with a single stone:

1. In this map function use this method
2. Fix the method, so that it will take the number of expected rent hours and then provide a correct estimation. By default this parameter can be set to 1 (1 hour).